partner_contact_lang/models/res_partner.py

```diff
@@ -21,21 +22,12 @@ def write(self, vals):
                 childs.write({'lang': vals['lang']})
         return res
 
-    @api.multi
-    def onchange_parent_id(self, parent_id):
+    @api.onchange('parent_id')
+    def onchange_parent_id(self):
         """Change language if the parent company changes and there's no
         language defined yet"""
-        res = super(ResPartner, self).onchange_parent_id(parent_id)
-        if parent_id and self.parent_id.id != parent_id and not self.lang:
-            parent = self.browse(parent_id)
+        res = super(ResPartner, self).onchange_parent_id()
+        if self.parent_id and self.parent_id.id != self.id and not self.lang:
```

Comment:
You can do `if self != self.parent_id and not self.lang` I think 🤔

Reply:
I can't. It must be checked before if parent_id exists or not.